pkg/querier/block_streaming.go

```diff
@@ -78,19 +85,31 @@ func (bqss *blockStreamingQuerierSeriesSet) Warnings() annotations.Annotations {
 }
 
 // newBlockStreamingQuerierSeries makes a new blockQuerierSeries. Input labels must be already sorted by name.
-func newBlockStreamingQuerierSeries(lbls labels.Labels, seriesIdxStart, seriesIdxEnd int, streamReader chunkStreamReader) *blockStreamingQuerierSeries {
+func newBlockStreamingQuerierSeries(lbls labels.Labels, seriesIdxStart, seriesIdxEnd int, streamReader chunkStreamReader, logger log.Logger, chunkInfo *chunkreplyformatter.ChunkReplyFormatter, lastOne bool, traceId, remoteAddress string) *blockStreamingQuerierSeries {
```

Comment:
To make this less of an invasive change, what if `chunkreplyformatter.ChunkReplyFormatter` held the logger and trace ID? Then we wouldn't need to pass these around everywhere.